src/vulkan-renderer/vk_tools/gpu_info.cpp

```diff
@@ -369,10 +369,10 @@ void print_physical_device_memory_properties(const VkPhysicalDevice graphics_car
 
         const auto &property_flag = gpu_memory_properties.memoryHeaps[i].flags;
 
-        if (property_flag & VK_MEMORY_HEAP_DEVICE_LOCAL_BIT) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_HEAP_DEVICE_LOCAL_BIT)) {
             spdlog::debug("VK_MEMORY_HEAP_DEVICE_LOCAL_BIT ");
         }
-        if (property_flag & VK_MEMORY_HEAP_MULTI_INSTANCE_BIT) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_HEAP_MULTI_INSTANCE_BIT)) {
             spdlog::debug("VK_MEMORY_HEAP_MULTI_INSTANCE_BIT");
         }
```

Comment:
When discussing those lines with you, you mentioned that "most of them shouldn't actually be static_casts", but `!= 0`.